packages/payment-detection/src/any-to-native-detector.ts

```diff
@@ -0,0 +1,29 @@
+import { ExtensionTypes, PaymentTypes } from '@requestnetwork/types';
+
+import { AnyToAnyDetector } from './any-to-any-detector';
+import { ReferenceBasedDetectorOptions } from './types';
+
+/**
+ * Handle payment detection for native token payment with conversion
+ */
+export abstract class AnyToNativeDetector extends AnyToAnyDetector<
+  ExtensionTypes.PnAnyToEth.IAnyToEth,
+  PaymentTypes.IETHPaymentEventParameters
+> {
+  public constructor({
```

Comment:
```suggestion
  protected constructor({
```

Reply:
done in https://github.com/RequestNetwork/requestNetwork/pull/950/commits/b3916cac36caa8784ae927e51f68410bba53f957

Reply:
I had to add some code in classes that extend it:
```ts
  constructor(args: NativeDetectorOptions) {
    super(args);
  }
```